packages/lib/src/vis/heatmap/HeatmapMesh.tsx

```diff
@@ -142,31 +141,34 @@ const SCALE_SHADER: Record<ScaleType, ScaleShader> = {
 
 function HeatmapMesh(props: Props) {
   const {
-    rows,
-    cols,
     values,
     domain,
-    colorMap,
     scaleType,
+    colorMap,
     invertColorMap = false,
+    textureType = TEXTURE_TYPE_BY_DTYPE[values.dtype],
     alphaValues,
     alphaDomain,
   } = props;
 
   const { ordinateConfig } = useAxisSystemContext();
 
   const dataTexture = useMemo(() => {
-    const valuesArr = Float32Array.from(values);
-    return new DataTexture(valuesArr, cols, rows, RedFormat, FloatType);
-  }, [cols, rows, values]);
+    const { data, shape } = values;
+    const format = getTextureFormatFromType(textureType);
+    return new DataTexture(data, shape[1], shape[0], format, textureType);
+  }, [textureType, values]);
```

Comment:
Perhaps it would be worth abstracting the hook/utility to construct a new `DataTexture` from a ndarray

Reply:
Good call, it's much cleaner. As discussed, I didn't go as far as creating a memoized hook so as to not have to handle the `undefined` case with function overrides...